src/librustc/mir/mod.rs

```diff
@@ -1154,6 +1153,21 @@ pub enum TerminatorKind<'tcx> {
     },
 }
 
+/// Information about an assertion failure.
+#[derive(Clone, RustcEncodable, RustcDecodable, HashStable, PartialEq)]
+pub enum PanicInfo<O> {
```

Comment:
wrt naming this thing, I think the pattern we commonly use is `*Kind`, so `AssertKind`?

Reply:
All right, `AssertKind` it is.